src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedItems: GroupsMap<T> = {};
+
+  for (let i = 0; i < items.length; i += 1) {
+    const item: T = items[i];
+    const groupBy: keyof GroupsMap<T> = String(item[key]);
```

Comment:
```suggestion
    const groupBy = String(item[key]);
```

Reply:
Key of GroupsMap<T> is string, so no need to specify this type here since you converted this value to string